src/components/_tests_/EnumSpan.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { shallow } from "enzyme";
+import { createElement } from "react";
+import * as classNames from "classnames";
+
+import { EnumSpan, EnumSpanProps } from "../EnumSpan";
+
+describe("EnumSpan", () => {
+    const renderEnumSpan = (props: EnumSpanProps) => shallow(createElement(EnumSpan, props));
+
+    const defaultProps: EnumSpanProps = {
+        enumAttributeValue: "Chicken",
+        status: "enabled",
+        onClickAction: jasmine.createSpy("onClick"),
+        caption: "Salami"
+    };
+
+    it("renders the structure correctly", () => {
+        const enumSpan = renderEnumSpan(defaultProps);
+
+        expect(enumSpan).toBeElement(
+            createElement("span", {
+                className: "span-default span-responsive",
+                onClick: defaultProps.onClickAction
+            }, defaultProps.caption)
+        );
+    });
+
+    it("should not have the class active when inactive", () => {
+        const enumSpan = renderEnumSpan(defaultProps);
+
+        expect(enumSpan.hasClass("active")).toBe(false);
+    });
+
+    it("should have the class active when active", () => {
+        defaultProps.caption = "Chicken";
```

Comment:
If you are sharing variables in unit tests. it should not change. If it changes create a copy and use in your tests